File: src/main/java/com/patres/timetable/domain/Interval.java
package com.patres.timetable.domain;

import org.hibernate.annotations.Cache;
import org.hibernate.annotations.CacheConcurrencyStrategy;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import java.io.Serializable;
import java.time.LocalDate;
import java.util.Objects;

@Entity
@Table(name = "interval")
@Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
public class Interval extends AbstractApplicationEntity implements Serializable {

    private static final long serialVersionUID = 6794788291603472570L;

    @Column(name = "included")
    private Boolean included;

    @Column(name = "start_date")
    private LocalDate startDate;

    @Column(name = "end_date")
    private LocalDate endDate;

    @ManyToOne
    private Period period;

    public Boolean isIncluded() {
        return included;
    }

    public Interval included(Boolean included) {
        this.included = included;
        return this;
    }

    public void setIncluded(Boolean included) {
        this.included = included;
    }

    public LocalDate getStartDate() {
        return startDate;
    }

    public void setStartDate(LocalDate startDate) {
        this.startDate = startDate;
    }

    public Interval startDate(LocalDate startDate) {
        this.startDate = startDate;
        return this;
    }

    public LocalDate getEndDate() {
        return endDate;
    }

    public void setEndDate(LocalDate endDate) {
        this.endDate = endDate;
    }

    public Interval endDate(LocalDate endDate) {
        this.endDate = endDate;
        return this;
    }

    public Period getPeriod() {
        return period;
    }

    public void setPeriod(Period period) {
        this.period = period;
    }

    public Interval period(Period period) {
        this.period = period;
        return this;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        Interval interval = (Interval) o;
        if (interval.getId() == null || getId() == null) {
            return false;
        }
        return Objects.equals(getId(), interval.getId());
    }

    @Override
    public int hashCode() {
        return Objects.hashCode(getId());
    }

    @Override
    public String toString() {
        return "Interval{" +
            "id=" + getId() +
            ", included='" + isIncluded() + "'" +
            ", startDate='" + getStartDate() + "'" +
            ", endDate='" + getEndDate() + "'" +
            "}";
    }
}

File: src/main/kotlin/com/patres/timetable/domain/Interval.kt
package com.patres.timetable.domain

import org.hibernate.annotations.Cache
import org.hibernate.annotations.CacheConcurrencyStrategy
import java.io.Serializable
import java.time.LocalDate
import java.util.*
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.ManyToOne
import javax.persistence.Table

@Entity
@Table(name = "interval")
@Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
class Interval(

    @Column(name = "included_state")
    var includedState: Boolean = true,

    @Column(name = "start_date")
    var startDate: LocalDate? = null,

    @Column(name = "end_date")
    var endDate: LocalDate? = null,

    @ManyToOne
    var period: Period? = null

) : AbstractApplicationEntity(), Serializable {

    override fun toString(): String {
        return "Interval{" +
            "id= '$id'" +
            ", includedState='$includedState" +
            ", startDate='$startDate'" +
            ", endDate='$endDate'" +
            "}"
    }

}
